shell/models/catalog.cattle.io.app.js

```diff
@@ -40,69 +41,48 @@ export default class CatalogApp extends SteveModel {
   }
 
   matchingChart(includeHidden) {
-    const chart = this.spec?.chart;
-
-    if ( !chart ) {
-      return;
-    }
-
-    const chartName = chart.metadata?.name;
-    const preferRepoType = chart.metadata?.annotations?.[CATALOG_ANNOTATIONS.SOURCE_REPO_TYPE];
-    const preferRepoName = chart.metadata?.annotations?.[CATALOG_ANNOTATIONS.SOURCE_REPO_NAME];
-    const match = this.$rootGetters['catalog/chart']({
-      chartName,
-      preferRepoType,
-      preferRepoName,
+    return this.$rootGetters['catalog/chart']({
+      chart: this.spec?.chart,
       includeHidden
     });
-
-    return match;
   }
 
   get currentVersion() {
     return this.spec?.chart?.metadata?.version;
   }
 
+  /**
+   * Return possible upgrades
+   * false = does not apply (managed by fleet)
+   * null = no upgrade found
+   * string = version available to upgrade to
+   */
   get upgradeAvailable() {
-    // false = does not apply (managed by fleet)
-    // null = no upgrade found
-    // object = version available to upgrade to
-
-    if (
-      this.spec?.chart?.metadata?.annotations?.[CATALOG_ANNOTATIONS.MANAGED] ||
-      this.spec?.chart?.metadata?.annotations?.[FLEET.BUNDLE_ID]
-    ) {
-      // Things managed by fleet shouldn't show upgrade available even if there might be.
+    // Things managed by fleet shouldn't show upgrade available even if there might be.
+    const isManaged = this.spec?.chart?.metadata?.annotations?.[CATALOG_ANNOTATIONS.MANAGED] ||
+    this.spec?.chart?.metadata?.annotations?.[FLEET.BUNDLE_ID];
+
+    if (isManaged) {
       return false;
     }
-    const chart = this.matchingChart(false);
 
-    if ( !chart ) {
+    const matchingChart = this.matchingChart(false);
+
+    if ( !matchingChart ) {
       return null;
     }
 
     const workerOSs = this.$rootGetters['currentCluster'].workerOSs;
-
     const showPreRelease = this.$rootGetters['prefs/get'](SHOW_PRE_RELEASE);
-
-    const thisVersion = this.spec?.chart?.metadata?.version;
-    let versions = chart.versions;
-
-    if (!showPreRelease) {
-      versions = chart.versions.filter((v) => !isPrerelease(v.version));
-    }
-
-    versions = compatibleVersionsFor(chart, workerOSs, showPreRelease);
-
-    const newestChart = versions?.[0];
-    const newestVersion = newestChart?.version;
-
-    if ( !thisVersion || !newestVersion ) {
```

Comment:
this check, for truthy currentChartVersion and newestVersion, has been lost. i noticed that `compare` now has 

```
if (in1 === undefined && in2 === undefined) {
    return null;
}
```
if that was meant for a replacement, would vote to keep `compare` untouched and to bring it back to this context (as compare is used in other places). So it would be

```
if (!currentChartVersion || !newestVersion) {
    return null;
}
```

Reply:
It fails if I keep as it is, I'll try to check what I can do.
Ideally would be convenient to have real data to understand better the suggestion.